Code/Framework/AzToolsFramework/AzToolsFramework/ToolsComponents/EditorComponentBase.cpp

```diff
@@ -94,5 +94,15 @@ namespace AzToolsFramework
         {
             return AzToolsFramework::IsSelected(GetEntityId());
         }
+
+        void EditorComponentBase::SetSerializedIdentifier(const AZStd::string& alias)
+        {
+            m_alias = alias;
+        }
```

Comment:
This should be accepted by value and moved into the m_alias member, to avoid an unnecessary copy when this function is invoked with a string literal.
i.e
```
MyEditorComponent editorComponent;
editorComponent->SetSerializedIdentifier("foo"); // Has to construct a new AzStd::string and bind it to a lvalue reference
// Then inside the function it must be make a copy because the parameter is const lvalue.
```
```suggestion
        void EditorComponentBase::SetSerializedIdentifier(AZStd::string alias)
        {
            m_alias = AZStd::move(alias);
        }
```